rfcs/20250612-libs-structure/README.md

```diff
@@ -0,0 +1,89 @@
+
+### Revision
+|Date       |Revision| Comments                                                                 |
+|-----------|--------|--------------------------------------------------------------------------|
+|  20250612 |  1.0   | Initial version                                                          |
+|  20250701 |  1.1   | Static sycl drop updates                                                 |
+
+## Motivation
+
+* What problem do we solve?
+  * **Significant reduction in library size**:  
+    Removing symbol copying for static libraries will reduce the size of the oneDAL standalone package by approximately:
+    - ~50% (~700 MB) on Linux
+    - ~X% (~X GB) on Windows (TBD)
+  * **Improved build and test efficiency**:  
+    Avoids long link times associated with large static archives, especially for internal testing and CI systems.
+  * **Alignment with Intel oneMKL roadmap**:  
+    Static SYCL libraries will be deprecated in upcoming oneMKL releases.  
+    Continuing support for them would require maintaining legacy configurations and diverging from upstream support.
+  * **Cleaner and more maintainable build system**:  
+    Simplifies both Bazel and CMake logic by removing fragile `addlib`-style manual bundling.
+
+* What is the customer impact if we don't do this?
+  * Users may face long build times, symbol conflicts, and compatibility issues with newer oneMKL versions.
+  * Continued dependency on deprecated SYCL static libraries will make integration harder with future toolchains.
+
+* What is the timeline?
+  * **Deprecation notice**: oneDAL 2025.7 release notes
+  * **Symbol copying removal**: oneDAL 2026.0
+  * **Static SYCL drop**: oneDAL 2026.0
+
+## Initial Results of Proposed Changes
+
+|Branch     | Size   | Building time                | DPC examples/test bazel linking           |
+|-----------|--------|------------------------------|-------------------------------------------|
+|   main    |  1.5gb | 4min 38s -j 112              | 55 s                                      |
+|   custom  |  710mb*| 3min 13s -j 112              | 15 s                                      |
+*may be changed in the final version
+
+## Introduction
+
+Currently, in the oneDAL build process, mathematical backends such as **OpenBLAS** for ARM and **Intel MKL** are linked by **copying all symbols** from the corresponding static libraries (e.g., `libopenblas.a`, `libmkl_core.a`) into the resulting archive via custom `addlib` logic in the `Makefile`.
+
+This approach leads to bundling large static libraries into our own library archives, increasing binary size and leading to symbol duplication or conflicts if the final application also links to the same libraries.
+
+In addition, the upcoming oneMKL releases will **remove support for static SYCL libraries** (`libmkl_sycl*.a`). To remain compatible with the future oneMKL toolchain and simplify our own build system, we propose to **drop static SYCL support in oneDAL** as well.
```

Comment:
i think we should separate proposals to make them clear - in fact i don't think there is need for SYCL one as currently it impact no other parties. 

The core gist of this proposal is idea of changing way how we handle static cases for oneDAL - instead of addlib math backends - define build time dependency on them